kroxylicious-integration-tests/src/test/java/io/kroxylicious/proxy/MetricsIT.java

```diff
@@ -152,6 +165,59 @@ void prometheusMetricsWithCommonTags(KafkaCluster cluster) {
         }
     }
 
+    @Test
+    void testKroxyliciousInboundDownstreamMessages(KafkaCluster cluster, Topic topic) {
+
+        NamedFilterDefinition namedFilterDefinition = new NamedFilterDefinitionBuilder("filter-1",
```

Comment:
I'd separate the test cases into two.  One for decoded message and one for the undecoded path.